src/sidebar/components/search-status-bar.js

```diff
@@ -1,63 +1,109 @@
+  const storeState = useStore(store => store.getState());
+  const onClearSelection = useStore(store => store.clearSelection);
+  const filterQuery = storeState.filterQuery;
+  const filterActive = !!storeState.filterQuery;
 
-  const thread = () => {
-    return rootThread.thread(store.getState());
-  };
+  const thread = rootThread.thread(storeState);
 
-  const visibleCount = memoize(thread => {
-    return thread.children.reduce(
+  const countVisibleAnns = annThread => {
+    return annThread.children.reduce(
       function(count, child) {
-        return count + visibleCount(child);
+        return count + countVisibleAnns(child);
       },
-      thread.visible ? 1 : 0
+      annThread.visible ? 1 : 0
     );
-  });
+  };
+
+  const visibleCount = useMemo(() => {
+    return countVisibleAnns(thread);
+  }, [thread]);
 
-  this.filterMatchCount = function() {
-    return visibleCount(thread());
+  const filterResults = () => {
+    const resultsCount = visibleCount;
+    switch (resultsCount) {
+      case 0:
+        return 'No results for ' + filterQuery;
+      case 1:
+        return '1 search result';
+      default:
+        return resultsCount + ' search results';
+    }
   };
 
-  this.areAllAnnotationsVisible = function() {
-    if (store.getState().directLinkedGroupFetchFailed) {
+  const areAllAnnotationsVisible = () => {
+    if (storeState.directLinkedGroupFetchFailed) {
       return true;
     }
-    const selection = store.getState().selectedAnnotationMap;
+    const selection = storeState.selectedAnnotationMap;
     if (!selection) {
       return false;
     }
     return Object.keys(selection).length > 0;
   };
 
-  this.filterQuery = function() {
-    return store.getState().filterQuery;
-  };
-
-  this.filterActive = function() {
-    return !!store.getState().filterQuery;
-  };
-
-  this.onClearSelection = function() {
-    store.clearSelection();
-  };
+  return (
+    <div className="search-status-bar">
+      {filterActive && (
+        <div className="search-status-bar">
+          <button
+            className="primary-action-btn primary-action-btn--short"
+            onClick={onClearSelection}
+            title="Clear the search filter and show all annotations"
+          >
+            <i className="primary-action-btn__icon h-icon-close"></i>Clear
+            search
+          </button>
+          <span>{filterResults()}</span>
+        </div>
+      )}
+      {!filterActive && areAllAnnotationsVisible() && (
+        <div className="search-status-bar">
+          <button
+            className="primary-action-btn primary-action-btn--short"
+            onClick={onClearSelection}
+            title="Clear the selection and show all annotations"
+          >
+            {selectedTab === uiConstants.TAB_ORPHANS && (
+              <Fragment>Show all annotations and notes</Fragment>
+            )}
+            {selectedTab === uiConstants.TAB_ANNOTATIONS && (
+                <Fragment>Show all annotations
```

Comment:
There is a missing space before the annotation count, and the same with the note count:

<img width="282" alt="Screenshot 2019-07-22 18 12 03" src="https://user-images.githubusercontent.com/2458/61650765-4dbb8c80-acac-11e9-86dd-759cafe71b87.png">